desc/compute/data_index.py

```diff
@@ -100,8 +101,12 @@ def register_compute_fun(
     parameterization : str or list of str
         Name of desc types the method is valid for. eg `'desc.geometry.FourierXYZCurve'`
         or `'desc.equilibrium.Equilibrium'`.
-    grid_type : str
-        Name of grid type the quantity must be computed with. eg `'quad'`.
+    resolution_requirement : str
+        Resolution requirements in coordinates. I.e. "r" expects radial resolution
+        in the grid, "rtz" expects grid to radial, poloidal, and toroidal resolution.
+    source_grid_requirement : dict
```

Comment:
might want a bit more explanation here. IE, the source grid is generally not the grid things are being computed on, right? since all of the regular grid types don't have a source grid by default.

Reply:
Yes. One way to think about it is the source grid of any `LinearGrid`, `QuadratureGrid`, `ConcentricGrid` is itself